src/util/util.cs

```diff
@@ -226,37 +226,52 @@ public static ProcessResult ExecuteProcess(string name, IEnumerable<string> comm
             }
 
             // Finally, start the process.
-            process.Start();
-
-            if (capture)
-            {
-                // Set up async reads for stderr and stdout
-                process.BeginErrorReadLine();
-                process.BeginOutputReadLine();
-            }
-            else
+            try
             {
-                string stdout = process.StandardOutput.ReadToEnd();
-                string stderr = process.StandardError.ReadToEnd();
+                process.Start();
 
-                if (!string.IsNullOrEmpty(stdout))
+                if (capture)
                 {
-                    Console.Write(stdout);
+                    // Set up async reads for stderr and stdout
+                    process.BeginErrorReadLine();
+                    process.BeginOutputReadLine();
                 }
-                if (!string.IsNullOrEmpty(stderr) && (stdout != stderr))
+                else
                 {
-                    Console.Write(stderr);
+                    string stdout = process.StandardOutput.ReadToEnd();
+                    string stderr = process.StandardError.ReadToEnd();
```

Comment:
I know this is preexisting code, but I wonder if there isn't potential for deadlock here if the child process writes more that can fit in its buffer into stderr.

Reply:
Based on the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.standardoutput?redirectedfrom=MSDN&view=net-5.0#System_Diagnostics_Process_StandardOutput), the answer appears to be "yes": it even gives this exact example.

I opened https://github.com/dotnet/jitutils/issues/344 to track.